apps/spruce/src/pages/waterfall/BuildRow.tsx

```diff
@@ -0,0 +1,161 @@
+import { memo, useCallback } from "react";
+import styled from "@emotion/styled";
+import { palette } from "@leafygreen-ui/palette";
+import { Link } from "react-router-dom";
+import { TaskStatus } from "@evg-ui/lib/types/task";
+import { useWaterfallAnalytics } from "analytics";
+import { StyledLink } from "components/styles";
+import { getTaskRoute, getVariantHistoryRoute } from "constants/routes";
+import { size } from "constants/tokens";
+import {
+  WaterfallBuild,
+  WaterfallBuildVariant,
+  WaterfallQuery,
+} from "gql/generated/types";
+import {
+  BuildVariantTitle,
+  gridGroupCss,
+  InactiveVersion,
+  Row,
+} from "./styles";
+
+const { black, gray, green, white } = palette;
+
+export const BuildRow: React.FC<{
+  build: WaterfallBuildVariant;
+  projectIdentifier: string;
+  versions: WaterfallQuery["waterfall"]["versions"];
+}> = ({ build, projectIdentifier, versions }) => {
+  const { sendEvent } = useWaterfallAnalytics();
+  const handleVariantClick = useCallback(
+    () => sendEvent({ name: "Clicked variant label" }),
+    [sendEvent],
+  );
+  const handleTaskClick = useCallback(
+    (status: string) => () =>
+      sendEvent({
+        name: "Clicked task box",
+        "task.status": status,
+      }),
+    [sendEvent],
+  );
+
+  const { builds, displayName } = build;
+  let buildIndex = 0;
+  return (
+    <Row>
+      <BuildVariantTitle>
+        <StyledLink
+          href={getVariantHistoryRoute(projectIdentifier, build.id)}
+          onClick={handleVariantClick}
+        >
+          {displayName}
+        </StyledLink>
+      </BuildVariantTitle>
+      <BuildGroup data-cy="build-group">
+        {versions.map(({ inactiveVersions, version }) => {
+          if (inactiveVersions) {
+            return <InactiveVersion data-cy="inactive-column" />;
+          }
+          // The list of builds returned does not include a placeholder for inactive builds, so we need to check whether the build matches the version in the current column
+          if (version && version.id === builds?.[buildIndex]?.version) {
```

Comment:
is the reason this works because the `builds` is guaranteed to be the same order as the `versions`? It could be good to leave a comment about that